docs/community/training/debugging/what-is-debugging.md

```diff
@@ -0,0 +1,82 @@
+# What is debugging?
+
+!!! info
+
+    Debugging is the process of identifying and removing errors from computer software.
+
+You need to **identify (and reproduce) the problem** and **only then** begin fixing it (ideally writing a test case first, to check that (a) you can identify the problem; and (b) to identify if you accidentally introduce the same, or similar, mistake in the future).
+
+## Action 1: Identify the error
+
+!!! tip
+
+    First make sure that you can reproduce the error.
+
+What observations or outputs indicate the presence of this error?
+
+Is the error reproducible, or does it come and go?
+
+Write a failing test?
```

Comment:
Oh we suggest the failing test here! I think adding it again at the point of my previous comment is also good.